src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -5083,8 +5083,11 @@ private List<DecoratedKey> keySamples(Iterable<ColumnFamilyStore> cfses, Range<T
      */
     private void startLeaving()
     {
-        Gossiper.instance.addLocalApplicationState(ApplicationState.STATUS_WITH_PORT, valueFactory.leaving(getLocalTokens()));
-        Gossiper.instance.addLocalApplicationState(ApplicationState.STATUS, valueFactory.leaving(getLocalTokens()));
+        DatabaseDescriptor.getSeverityDuringDecommission()
+                          .ifPresent(severity ->
+                                     Gossiper.instance.addLocalApplicationState(ApplicationState.SEVERITY, valueFactory.severity(severity)));
```

Comment:
Why not call DES.setSeverity()?

Reply:
at first I didn't do it as `org.apache.cassandra.locator.DynamicEndpointSnitch#setSeverity` is a method and not function (tied to the current snitch).  I could always make it static.    To avoid copy/paste ill make it static and delegate to it

Reply:
pushed a commit fixing this.  see https://github.com/apache/cassandra/pull/2261/commits/f99bd8b1ef940ed2cdafc7b8af91e6ab1173937e

`setSeverity` is part of the `MBean` so I can not touch.  I also can not have a static method with the same name, so I created a new `addSeverity` static function, and have `setSeverity` and this logic call it